File: SDis/p2/Client-Java/src/main/java/pt/ulisboa/tecnico/tuplespaces/client/ClientMain.java
package pt.ulisboa.tecnico.tuplespaces.client;

import pt.ulisboa.tecnico.tuplespaces.client.grpc.ClientService;

public class ClientMain {

    public static void main(String[] args) {

        System.out.println(ClientMain.class.getSimpleName());

        // check arguments
        if (args.length != 2 && args.length != 3) {
            System.err.println("Invalid number of arguments!");
            System.err.println("Usage: mvn exec:java -Dexec.args=<host:port> <client_id> [-debug]");
            return;
        }

        // get the host and the port of the server or front-end
        final String host_port = args[0];
        final int client_id = Integer.parseInt(args[1]);

        // check if debug mode is enabled
        final boolean DEBUG = (args.length == 3 && args[2].equals("-debug"));

        if (DEBUG) {
            System.err.println("[\u001B[34mDEBUG\u001B[0m] Debug mode enabled");

            // receive and print arguments
            System.err.printf("[\u001B[34mDEBUG\u001B[0m] Received %d arguments%n", args.length);
            for (int i = 0; i < args.length; i++) {
                System.err.printf("[\u001B[34mDEBUG\u001B[0m] arg[%d] = %s%n", i, args[i]);
            }
        }

        // create the parser and start the command line interface
        CommandProcessor parser = new CommandProcessor(new ClientService(host_port, client_id, DEBUG));
        parser.parseInput();

    }
}

File: SDis/p2/Client-Python/client_main.py
import sys
from typing import List
from client_service import ClientService
from command_processor import CommandProcessor

class ClientMain:
    @staticmethod
    def main(args: List[str]):
        print("Python Client")

        if len(args) != 2 and len(args) != 3:
            print("Argument(s) missing!", file=sys.stderr)
            print("Usage: python3 client_main.py <host:port> <client_id> [-debug]", file=sys.stderr)
            return

        DEBUG = len(args) == 3 and args[2] == "-debug"

        if DEBUG:
            print("[\u001B[34mDEBUG\u001B[0m] Debug mode enabled")

            # receive and print arguments
            print(f"[\u001B[34mDEBUG\u001B[0m] Received {len(args)} arguments", file=sys.stderr)
            for i, arg in enumerate(args):
                print(f"[\u001B[34mDEBUG\u001B[0m] arg[{i}] = {arg}", file=sys.stderr)

        # Get the host and port of the server or front-end
        host_port = args[0]
        client_id = args[1]

        parser = CommandProcessor(ClientService(host_port, client_id, DEBUG))
        parser.parse_input()


if __name__ == "__main__":
    ClientMain.main(sys.argv[1:])